agents/utils.py

```diff
@@ -423,6 +423,26 @@ def perform_action(state, action):
     state['is_player_turn'] = not state['is_player_turn']
     state.get('action_history', []).append(action_event)
 
+def undo_last_action_mcts(state, action):
```

Comment:
Do we need to pass in the action? cant we just undo the last on in the history?

Reply:
where is the history stored?